### PR TITLE
Carbon application deployer for dashboard themes

### PR DESCRIPTION
**Pull request by prabushi**
Allow user to deploy themes using a .car file, and those deployed themes are used to allow users to choose a custom theme per dashboard.

### CODE REVIEW

File: apps/portal/theme/templates/dashboard-settings.jag [hunk 1]
```diff
@@ -15,7 +15,8 @@
  * limitations under the License.
  */
 %>
-<% include(utils.resolvePath('templates/portal-header.jag')); %>
+<% var utils = require('/modules/utils.js'); %>
```
Comment (sinthuja) [inline apps/portal/theme/templates/dashboard-settings.jag]: Code formatting issue.

Comment (prabushi) [inline apps/portal/theme/templates/dashboard-settings.jag]: Resolved.

File: apps/portal/modules/utils.js [hunk 2]
```diff
@@ -156,26 +198,18 @@ var dashboardLayouts = function () {
...
-    var file = new File('/' + path);
-    return file.isExists() ? path : null;
-};
-
-var portalStyles = function () {
+var getScript = function (fileName) {
     var config = require('/configs/designer.json');
     var theme = config.theme;
-    var path = 'extensions/themes/' + theme + '/css/portal.css';
+    var path = 'extensions/themes/' + theme + '/js/' + fileName + '.js';
```
Comment (sinthuja) [inline apps/portal/modules/utils.js]: Please use this as constants - extensions/themes/

File: apps/portal/modules/utils.js [hunk 3]
```diff
@@ -233,3 +267,28 @@ var resolvePassword = function(passwordAlias){
...
+ * @param dashboardTheme dashboard theme name
+ * @returns {String} dashboard theme
+ */
+var getDashboardTheme = function (dashboardThemeName) {
+    var themePath = getCustomThemePath() + dashboardThemeName;
+    var folder = new File(themePath);
+    if(dashboardThemeName !== undefined || folder.isExists()) {
+        return dashboardThemeName;
+    } else {
+        return "Default Theme";
```
Comment (sinthuja) [inline apps/portal/modules/utils.js]: Why not return the constant that has been defined?

File: apps/portal/modules/utils.js [hunk 4]
```diff
@@ -233,3 +267,28 @@ var resolvePassword = function(passwordAlias){
...
     return reslover.resolve(alias[1]); 
 };
+
+/**
+ * Returns the custom path after appending the current user domain
+ * @returns {String} path to custom themes
+ */
+var getCustomThemePath = function () {
+    var carbon = require('carbon');
+    return '/store/' + carbon.userDomain + '/fs/themes/';
```
Comment (sinthuja) [inline apps/portal/modules/utils.js]: Please use constants here too.

File: apps/portal/extensions/locales/locale_si.json [hunk 5]
```diff
@@ -139,4 +139,6 @@
   "designer.componentProperties.priority": "පූර්වතාව",
   "bad.request" : "වැරදි ඉල්ලීමකි",
   "forbidden" : "Forbiden"
```
Comment (Niranjan-K) [inline apps/portal/extensions/locales/locale_si.json]: This is corrected in the above file but not here. Shouldn't this be in Sinhalese and not in English?

Comment (prabushi) [inline apps/portal/extensions/locales/locale_si.json]: Fixed in commit 2c68f18dc9b0dc8bfdba57b8f658b2f55d199e99
https://github.com/wso2/carbon-dashboards/pull/452/commits/2c68f18dc9b0dc8bfdba57b8f658b2f55d199e99#diff-b5bb95287d2a3bc59e75146a3ab74c38R147

**Pull request merged**